### PR TITLE
CQ-4350702 Fix form submission failing in Sites

### PR DESCRIPTION
**Pull request by ujjwal5**
## Description

Currently form submission is failing in Sites since rest APIs expects guide container to be located at `/content/forms/af/<form-name>/jcr:content/guideContainer`
However, in a Sites page this is not guaranteed.

With this PR, we will be sending the encoded guide container path to the rest APIs instead of just sending the encoded path of the page/form.

This PR will help in fixing following bugs:

- Fix form submit in Sites.
- In case of multiple form containers, we should always show thank you page/message for which submit is clicked. This PR fixes that. We will be fetching form container using path instead of data.
- currently we always show default thank you page even if we have selected thank you message. This PR fixes that. Exposing thankYouOption/redirectUrl for this, which will do the business logic check in rest APIs.

Note: Skipping submit and prefill tests till we have updated changes for af2-rest-api in aemds.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java
Comment (nit23uec) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java:183]: we cannot merge this as it'll break the next release. pls have a logic to deduce if form has been authored from sites editor.

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java
Comment (nit23uec) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java:103]: thank you option is not part of schema.

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/models/form/FormComponent.java
Comment (nit23uec) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/models/form/FormComponent.java:82]: this will be taken care of @devgurjar . we are introducing a custom prop `fd:path`.

File: ui.frontend/src/utils.js
Comment (nit23uec) [inline ui.frontend/src/utils.js:200]: this is not correct. we should not decode the id at client.

File: ui.tests/test-module/specs/actions/submit/submit.runtime.spec.js
Comment (nit23uec) [inline ui.tests/test-module/specs/actions/submit/submit.runtime.spec.js:53]: this should not be skipped.

File: ui.tests/test-module/specs/prefill/customprefill.spec.js
Comment (nit23uec) [inline ui.tests/test-module/specs/prefill/customprefill.spec.js:55]: should not be skipped

**Review by nit23uec — CHANGES_REQUESTED** [review]
inline comments.

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java
Comment (ujjwal5) [reply bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java:103]: how do we handle if author wants to show message then in v2 on form submission?

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java
Comment (ujjwal5) [reply bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java:183]: added a check in getAction() to check if forms created in sites

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/models/form/FormComponent.java
Comment (ujjwal5) [reply bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/models/form/FormComponent.java:82]: added minimum requirement to unblock myself

File: ui.tests/test-module/specs/actions/submit/submit.runtime.spec.js
Comment (ujjwal5) [reply ui.tests/test-module/specs/actions/submit/submit.runtime.spec.js:53]: added back after adding condition

File: ui.tests/test-module/specs/prefill/customprefill.spec.js
Comment (ujjwal5) [reply ui.tests/test-module/specs/prefill/customprefill.spec.js:55]: added back after adding condition

File: ui.frontend/src/utils.js
Comment (ujjwal5) [reply ui.frontend/src/utils.js:200]: currently there's no other way to provide the encoded data for Sites page, since id will be different now

**Comment by nit23uec** [comment]
pls have unit tests for the APIs.

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java
Comment (rismehta) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java:171]: I see this code at multiple places. Can you create a utility for this and re-use it everywhere ?

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java
Comment (rismehta) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java:159]: Would we be supporting both path's (ie) form path and form container path ?

File: bundles/af-core/src/test/resources/form/formcontainer/test-forms-in-sites.json
Comment (rismehta) [inline bundles/af-core/src/test/resources/form/formcontainer/test-forms-in-sites.json:28]: You won't have fd:translationIds when a form is used inside sites ? Why do you need this

Comment (ujjwal5) [reply bundles/af-core/src/test/resources/form/formcontainer/test-forms-in-sites.json:28]: yes, used an existing json hence it is appearing, not using this in the tests for sites.

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java
Comment (ujjwal5) [reply bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java:159]: right now we are only differentiating based on forms and sites, from next release these checks will go away and we will only be having form container path and sites container path.

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java
Comment (ujjwal5) [reply bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java:157]: done

File: bundles/af-core/src/test/resources/form/formcontainer/test-forms-in-sites.json
Comment (ujjwal5) [reply bundles/af-core/src/test/resources/form/formcontainer/test-forms-in-sites.json]: removed

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/container.html
Comment (nit23uec) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/container.html:31]: I don't see a getRedirect method in the interface.

Comment (ujjwal5) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/container.html]: changed to redirectUrl. Thanks for pointing it out!

**Pull request merged**
